sdk/core/core-lro/test/lro.spec.ts

```diff
@@ -2702,6 +2877,264 @@ matrix(
+                },
+              ],
+              throwOnNon2xxResponse,
+            });
+            const await1 = await poller.then((result) => {
+              assert.equal(result.statusCode, 200);
+              return result;
+            });
+            const await2 = await poller.then((result) => {
+              assert.equal(result.statusCode, 200);
+              return result;
+            });
+            assert.isTrue(await1 === await2);
+          });
+          it("should trigger the whole polling process to server side only once", async () => {
+            let pollCount = 0;
+            const pollingPath = "pollingPath";
+            const poller = createTestPoller({
+              routes: [
+                {
+                  method: "POST",
+                  status: 202,
+                  headers: {
+                    "Operation-Location": pollingPath,
+                  },
+                },
+                ...Array(10).fill({
+                  method: "GET",
+                  path: pollingPath,
+                  body: `{ "status": "running" }`,
+                  status: 200,
+                }),
+                {
+                  method: "GET",
+                  path: pollingPath,
+                  body: `{ "status": "succeeded" }`,
+                  status: 200,
+                },
+              ],
+              implName,
+              throwOnNon2xxResponse,
+              updateState: () => {
+                pollCount++;
+              },
+            });
+            await poller;
+            await poller;
+            await poller;
+            assert.equal(pollCount, 11);
+          });
+          it("should catch the same error in multiple times", async () => {
+            const body = { status: "canceled", results: [1, 2] };
+            const errMsg = "Operation was canceled";
+            const pollingPath = "/LROPostDoubleHeadersFinalAzureHeaderGetDefault/asyncOperationUrl";
+            const poller = createTestPoller({
+              routes: [
+                {
+                  method: "POST",
+                  status: 202,
+                  headers: {
+                    location: "/postasync/retry/succeeded/operationResults/foo/200/",
+                    ["Operation-Location"]: pollingPath,
+                    "retry-after": "0",
+                  },
+                  body: `{ "properties": { "provisioningState": "Accepted"}, "id": "100", "name": "foo"}`,
+                },
+                {
+                  method: "GET",
+                  path: pollingPath,
+                  status: 200,
+                  body: JSON.stringify(body),
+                },
+              ],
+              implName,
+              throwOnNon2xxResponse: true,
+            });
+            try {
+              await poller;
+              assert.fail("should throw error");
+            } catch (e: any) {
+              assert.equal(e.message, errMsg);
+            }
```

Comment:
I think this can be simplified as `await assert.rejected(poller, errMsg);

Reply:
updated